tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -481,26 +481,35 @@ private:
          TVirtualCollectionProxy::DeleteTwoIterators_t fDeleteTwoIterators;
          TVirtualCollectionProxy::Next_t fNext;
       };
-      static RIteratorFuncs GetIteratorFuncs(TVirtualCollectionProxy *proxy, bool readOnly);
+      static RIteratorFuncs GetIteratorFuncs(TVirtualCollectionProxy *proxy, bool readFromDisk);
+
    private:
       class RIterator {
          const RCollectionIterableOnce &fOwner;
          void *fIterator = nullptr;
          void *fElementPtr = nullptr;
+
+         void Advance()
+         {
+            auto fnNext_Contig = [&]() {
+               auto &iter = *static_cast<unsigned char **>(fIterator), p = iter;
+               iter += fOwner.fStride;
+               return p;
+            };
+            fElementPtr = fOwner.fStride ? fnNext_Contig() : fOwner.fIFuncs.fNext(fIterator, fOwner.fEnd);
```

Comment:
I think this would be more readable with an `if (fOwner.fStride) { /* ... */  } else { /*...*/ }` block and without the lambda.

Reply:
Thanks for the review!  I initially thought that this form would be more suitable for the compiler optimizer, but I checked and both forms generate the same code, at least on clang 13 and GCC 13.

Other than that, might be personal taste, but I find it also more expressive, i.e. it reads as "assign the current element pointer from either (i) a simple increment, if the stride is known (e.g. for an array), or (ii) fallback to calling `fNext()`.  Thus, I'd vote to leave as-is.